e2e/docker-compose.yml

```diff
@@ -1,26 +1,21 @@
 version: "3.7"
 services:
-  streamlit:
-    image: streamlit_circleci
-    build: .
-    ports:
-      - "3000:3000"
```

Comment:
Is this port mapping is redundant in the previous version?

Reply:
The port mapping here is actually used to expose port 3000 of the docker container to the local machine, and I decided that it's most likely annoying to have to stop the local dev server that people normally have already-running on port 3000 when taking snapshots

I think ideally, we would want the docker container to be able to access the host machine's react dev server (it currently has to spin up its own when running tests), but that ends up being annoying to get working in both MacOS and Linux because

* you can only use `localhost` in the container to access the host machine in Linux and not MacOS
* on MacOS, you can access the special DNS name `host.docker.internal` within a container to talk to the host, but this DNS name doesn't work on Linux without passing some additional flags
* We would have to change all of our e2e tests to use either `localhost` or `host.docker.internal` depending on the environment the test is being run in

This might be something we want to smooth out later when there's time to do so, but for now totally isolating the container's dev server seems okay to me